code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -459,7 +454,7 @@ private static Map<String, Object> getXdmData(final String eventType, final Stri
         final Map<String, Object> trackingMap = new HashMap<>();
         final Map<String, Object> customActionMap = new HashMap<>();
 
-        if (actionId != null) {
+        if (!StringUtils.isNullOrEmpty(actionId)) {
```

Comment:
is empty string valid?

Reply:
double checked the ios implementation and an empty string is valid. I'll revert this change.

Reply:
looking into this further, imo an empty string is not valid. the action id is extracted in this line:
```java
final String actionId = eventData.optString(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_ACTION_ID, null);
```
with the check above previously checking for null string only. i think updating it to using the StringUtils method is not necessary as the value will always be a valid string or null.